torch/csrc/api/include/torch/serialize.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <torch/nn/module.h>
+#include <torch/nn/pimpl.h>
+#include <torch/serialize/archive.h>
+#include <torch/tensor.h>
+
+#include <string>
+#include <utility>
+
+namespace torch {
+namespace optim {
+class Optimizer;
+} // namespace optim
+} // namespace torch
+
+namespace torch {
+namespace serialize {
+template <typename ModuleType>
+void save(const nn::ModuleHolder<ModuleType>& module, OutputArchive& archive) {
+  module->save(archive);
+}
+
+template <typename ModuleType>
+void load(nn::ModuleHolder<ModuleType>& module, InputArchive& archive) {
+  module->load(archive);
+}
+
+void save(const Tensor& tensor, OutputArchive& archive);
+void load(Tensor& tensor, InputArchive& archive);
```

Comment:
Can we please not have mutable references in this API. Just pass in a `const Tensor&`, it's just as good (unless you're actually bonking in a new pointer value for Tensor, in which case just return the Tensor here!)